jumpcloud_bootstrap_template.sh

```diff
@@ -987,39 +1142,283 @@ if [[ ! -f $DEP_N_GATE_DONE ]]; then
+            ORG_ID_RAW="${BASH_REMATCH[@]}"
+        fi
+
+        ORG_ID=$(echo $ORG_ID_RAW | cut -d '"' -f 4)
+        # Get ORG ID
+        if [[ -z $ORG_ID ]]; then
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ORGID Variable is missing" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): Getting System Details with systemContext API" >>"$DEP_N_DEBUG"
+
+            # Get the current time.
+            now=$(date -u "+%a, %d %h %Y %H:%M:%S GMT")
+            # create the string to sign from the request-line and the date
+            signstr="GET /api/systems/${systemID} HTTP/1.1\ndate: ${now}"
+
+            # create the signature
+            signature=$(printf "$signstr" | openssl dgst -sha256 -sign /opt/jc/client.key | openssl enc -e -a | tr -d '\n')
+
+            # make the api call passing the signature in the authorization header
+            httpRequest=$(
+                curl -s -o $DEP_N_HTTP_RESPONSE -w "%{http_code}" \
+                    -H "Accept: application/json" \
+                    -H "Date: ${now}" \
+                    -H "Authorization: Signature keyId=\"system/${systemID}\",headers=\"request-line date\",algorithm=\"rsa-sha256\",signature=\"${signature}\"" \
+                    --url https://console.jumpcloud.com/api/systems/${systemID}
+                )
+            # Check Response
+            responseContent=$(cat $DEP_N_HTTP_RESPONSE)
+            if [[ $httpRequest != "200" ]]; then
+                # handle error
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): [error] HTTP Response does not indicate success" >>"$DEP_N_DEBUG"
+                # echo "$(date "+%Y-%m-%d %H:%M:%S"): HTTP Response: $responseContent"
+            else
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): System details returned successfully" >>"$DEP_N_DEBUG"
+                # echo "$(date "+%Y-%m-%d %H:%M:%S"): HTTP Response: $responseContent"
+                # Get ORGID
+                regexORG='\"organization\":\"([a-zA-Z0-9]{24})\"'
+                if [[ $responseContent =~ $regexORG ]]; then
+                    ORG_ID="${BASH_REMATCH[1]}"
+                fi
+                APIKEY=$(DecryptKey $ENCRYPTED_KEY $DECRYPT_USER_ID $ORG_ID)
+            fi
+        fi
+        # check if we can validate the APIKEY (this may not be possible in case of a reboot)
+        if [[ -n $ENCRYPTED_KEY && -n $DECRYPT_USER_ID  && -n $ORG_ID ]]; then
+            APIKEY=$(DecryptKey $ENCRYPTED_KEY $DECRYPT_USER_ID $ORG_ID)
+        else
+        echo "$(date "+%Y-%m-%d %H:%M:%S"): One of the required variables is null, script will progress and use the system context API Key" >>"$DEP_N_DEBUG"
+            SelfSignedRequest=true
+        fi
+    fi
+    # get the enrollment user if not there:
+    if [[ -z $enrollmentUser ]]; then
+        enrollmentUser=$(cat $DEP_N_PERSIST)
+    fi
+
+    echo "$(date "+%Y-%m-%d %H:%M:%S"): Waiting for $enrollmentUser to login before we can continue." >>"$DEP_N_DEBUG"
```

Comment:
the code should validate that $enrollmentUser is logged in, not just wait for Finder Process